src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2AuthenticationContext.cs

```diff
@@ -122,7 +122,7 @@ public Uri DeclarationReference
             set
             {
                 if (value != null && !value.IsAbsoluteUri)
-                    throw LogExceptionMessage(new ArgumentException(FormatInvariant(LogMessages.IDX13300, nameof(value), value)));
+                    throw LogExceptionMessage(new ArgumentException(FormatInvariant(LogMessages.IDX13300, nameof(DeclarationReference), value)));
```

Comment:
@keegan-caruso could you fix up the difference between ClassReference and DeclarationReference where ClassReference throws ArgumentNullException when value == null. I would vote for differentiating null.

Reply:
@brentschmaltz I'm happy to normalize the two. I am wondering if this goes back to the Saml2Core spec. Section 2.7.2.2 calls out both as being optional. That seems to imply neither throw on null. This looks safe as all access to this field checks for null conditions.

Reply:
@keegan-caruso in rethinking this, the ctors seem to be fine. Given the spec.
The issue seems to be that the ctors try and set a null value to the property.

It's OK that the value is null, but the reason we throw when setting the property is that seems to indicate a programming error.

So, perhaps the error is the constructor trying to set a null value.

Reply:
This is a good idea. It also fixes #1457